test/ruby/test_weakmap.rb

```diff
@@ -16,16 +16,16 @@ def test_map
 
   def test_aset_const
     x = Object.new
-    assert_raise(ArgumentError) {@wm[true] = x}
-    assert_raise(ArgumentError) {@wm[false] = x}
-    assert_raise(ArgumentError) {@wm[nil] = x}
-    assert_raise(ArgumentError) {@wm[42] = x}
-    assert_raise(ArgumentError) {@wm[:foo] = x}
-    assert_raise(ArgumentError) {@wm[x] = true}
-    assert_raise(ArgumentError) {@wm[x] = false}
-    assert_raise(ArgumentError) {@wm[x] = nil}
-    assert_raise(ArgumentError) {@wm[x] = 42}
-    assert_raise(ArgumentError) {@wm[x] = :foo}
+    @wm[true] = x
+    @wm[false] = x
+    @wm[nil] = x
+    @wm[42] = x
+    @wm[:foo] = x
+    @wm[x] = true
+    @wm[x] = false
+    @wm[x] = nil
+    @wm[x] = 42
+    @wm[x] = :foo
```

Comment:
Instead of just removing assertions all, could you add `assert_same(x, @wm[true])` and so on after each assignment?